### ISSUE TITLE
SC1073 false positive when directive comment at the end of a case statement

### ISSUE THREAD

**Issue by vapier** [koalaman/shellcheck#1127]
#### For bugs
using online and 0.4.6 locally to same result

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
case ${COMMAND} in
*)
  # shellcheck disable=SC2086
  ${COMMAND}
  # shellcheck disable=SC2086
  ;;
esac
```

#### Here's what shellcheck currently says:

```
Line 2:
case ${COMMAND} in
^-- SC1009: The mentioned syntax error was in this case expression.
 
Line 3:
*)
^-- SC1073: Couldn't parse this case item. Fix to allow more checks.
 
Line 6:
  # shellcheck disable=SC2086
  ^-- SC1123: ShellCheck directives are only valid in front of complete compound commands, like 'if', not e.g. individual 'elif' branches.
 
Line 7:
  ;;
    ^-- SC1072: Unexpected keyword/token. Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

shouldn't be any errors

**Comment by koalaman** [comment]
This was by design, but the design should probably have been different. What's the expected scope of this directive?

**Comment by vapier** [comment]
the version Google is using has enable support, so someone was putting an enable statement at the end.  since the directive was correct, it was left with the SC1073 which is confusing.

as it is, shellcheck is emitting 4 warnings here and it's not clear that fixing the 3rd one would make the other 3 fix themselves too.  usually people start with the first error and work their way down which doesn't work in this case.

**Comment by vanniktech** [comment]
I'm also hitting this with the following:

```shell
# shellcheck.
sudo apt-get install shellcheck
```

Is there any timeline for a fix? Currently I workaround by uppercasing the `s`.

**Comment by nthexwn** [comment]
I inadvertently triggered [SC1073](https://github.com/koalaman/shellcheck/wiki/SC1073), [SC1072](https://github.com/koalaman/shellcheck/wiki/SC1072), and [SC1009](https://github.com/koalaman/shellcheck/wiki/SC1009) by placing a directive in the following code:

    if [[ ! -z "$1" ]]; then
        log_level="$1"
    # LOG_LEVEL is an optional external configuration, so it's okay for it to be unset.
    # shellcheck disable=SC2153
    elif [[ ! -z "$LOG_LEVEL" ]]; then
        log_level="$LOG_LEVEL"
    else
        log_level="info"
    fi

I believe it would have been more helpful to have [SC1123](https://github.com/koalaman/shellcheck/wiki/SC1123) shown here.  After seeing its description in the OP's output I was able to resolve this by moving the comments in my code to before the whole 'if' statement:

    # LOG_LEVEL is an optional external configuration, so it's okay for it to be unset.
    # shellcheck disable=SC2153
    if [[ ! -z "$1" ]]; then
        log_level="$1"
    elif [[ ! -z "$LOG_LEVEL" ]]; then
        log_level="$LOG_LEVEL"
    else
        log_level="info"
    fi

It would be great if Shellcheck could parse directives at arbitrary points within commands, but without diving into the parsing engine I can only conjecture about how difficult this might be.

**Comment by vapier** [comment]
another example but not using `case`:
```sh
#!/bin/bash
foo() {
  # shellcheck disable=SC2086
  $1 "$2"
  # shellcheck enable=SC2086
}
```

shellcheck.net reports:
```
$ shellcheck myscript
 
Line 2:
foo() {
^-- SC1009: The mentioned syntax error was in this function.
      ^-- SC1073: Couldn't parse this brace group. Fix to allow more checks.
 
Line 5:
  # shellcheck enable=SC2086
  ^-- SC1123: ShellCheck directives are only valid in front of complete compound commands, like 'if', not e.g. individual 'elif' branches.
 
Line 6:
}
 ^-- SC1072: Unexpected keyword/token. Fix any mentioned problems and try again.
```
